src/controller/event/handler/internal/artifact.go

```diff
@@ -240,3 +246,45 @@ func (a *Handler) onPush(ctx context.Context, event *event.ArtifactEvent) error
 
 	return nil
 }
+
+func isScannerUser(ctx context.Context, operator string, projectID int64, getPrefix scanRobotPrefixFunc) bool {
+	if len(operator) == 0 {
+		return false
+	}
+	robotPrefix := config.RobotPrefix(ctx)
+	if !strings.HasPrefix(operator, robotPrefix) {
+		return false
+	}
+	prefix, err := getPrefix(ctx, pkg.ProjectMgr, scanner.DefaultController, projectID)
+	if err != nil {
+		log.Debugf("failed to get scanner robot account prefix for project %d, error: %v", projectID, err)
+		return false
+	}
+	return strings.HasPrefix(operator, prefix)
+}
+
+type scanRobotPrefixFunc func(ctx context.Context, projManager project.Manager, scannerCtl scanner.Controller, projectID int64) (string, error)
+
+// scanRobotPrefix get the scanner robot account prefix by project id
+func scanRobotPrefix(ctx context.Context, projManager project.Manager, scannerCtl scanner.Controller, projectID int64) (string, error) {
```

Comment:
we should make sure that the creation and validation are using the same func to get the prefix.

Reply:
ok. add uuid in prefix